src/devices/Ht1632/Ht1632.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+
+namespace Iot.Device.Ht1632
+{
+    /// <summary>
+    /// HT1632C
+    /// 32x8 and 24x16 LED Driver
+    /// </summary>
+    public class Ht1632 : IDisposable
+    {
+        // Datasheet: https://www.holtek.com/documents/10179/116711/HT1632D_32D-2v100.pdf
+
+        /// <summary>
+        /// Set if system oscillator is on. (default is off)
+        /// </summary>
+        public bool Enabled { set => WriteCommand(value ? Command.SYS_EN : Command.SYS_DIS); }
+
+        /// <summary>
+        /// Set if LED duty cycle generator is on (default is off)
+        /// </summary>
+        public bool LedOn { set => WriteCommand(value ? Command.LED_On : Command.LED_Off); }
+
+        /// <summary>
+        /// Set if blinking function is on (default is off)
+        /// </summary>
+        public bool Blink { set => WriteCommand(value ? Command.BLINK_On : Command.BLINK_Off); }
+
+        /// <summary>
+        /// Set clock mode (default is RC-Master)
+        /// </summary>
+        public ClockMode ClockMode { set => WriteCommand((Command)value); }
+
+        /// <summary>
+        /// Set COM option (default is N-MOS open drain output and 8 COM)
+        /// </summary>
+        public ComOption ComOption { set => WriteCommand((Command)value); }
+
+        /// <summary>
+        /// Set row PWM duty (1/16 to 16/16, default is 16)
+        /// </summary>
+        public byte PwmDuty { set => WriteCommand(s_pwmCommandMap[value]); }
+
+        private static readonly Dictionary<byte, Command> s_pwmCommandMap = new Dictionary<byte, Command>()
+        {
+            { 1, Command.PWM_Duty_1 },
+            { 2, Command.PWM_Duty_2 },
+            { 3, Command.PWM_Duty_3 },
+            { 4, Command.PWM_Duty_4 },
+            { 5, Command.PWM_Duty_5 },
+            { 6, Command.PWM_Duty_6 },
+            { 7, Command.PWM_Duty_7 },
+            { 8, Command.PWM_Duty_8 },
+            { 9, Command.PWM_Duty_9 },
+            { 10, Command.PWM_Duty_10 },
+            { 11, Command.PWM_Duty_11 },
+            { 12, Command.PWM_Duty_12 },
+            { 13, Command.PWM_Duty_13 },
+            { 14, Command.PWM_Duty_14 },
+            { 15, Command.PWM_Duty_15 },
+            { 16, Command.PWM_Duty_16 },
+        };
+
+        private readonly Ht1632PinMapping _pinMapping;
+        private readonly int _cs;
+        private readonly int _wr;
+        private readonly int _data;
+        private readonly bool _shouldDispose;
+
+        private GpioController? _controller;
```

Comment:
Why do you want to have it nullable while you'll set it up in the constructor? You're checking that it's not null and create one if it is. so a private readonly should work as well  here without the nullability.